src/Modules/UngradedModules.jl

```diff
@@ -656,36 +790,37 @@ Construct the morphism $F \to G$ corresponding to the matrix `mat`.
 FreeModuleHom(F::FreeMod{T}, G::S, mat::MatElem{T}) where {T,S} = FreeModuleHom{T,S}(F, G, mat)
 
 @doc Markdown.doc"""
-    matrix(f::FreeModuleHom)
+    matrix(a::FreeModuleHom)
 
-Compute the matrix corresponding to the morphism `f`.
-The matrix is cached.
+Return the matrix corresponding to `a`.
+
+If the matrix is not yet cached, compute and cache the matrix first.
 """
-function matrix(f::FreeModuleHom)
-  if !isdefined(f, :matrix)
-    D = domain(f)
-    C = codomain(f)
+function matrix(a::FreeModuleHom)
+  if !isdefined(a, :matrix)
+    D = domain(a)
+    C = codomain(a)
     R = base_ring(D)
     matrix = zero_matrix(R, rank(D), ngens(C))
     for i=1:rank(D)
-      image_of_gen = f(D[i])
+      image_of_gen = a(D[i])
       for j=1:ngens(C)
         matrix[i,j] = image_of_gen[j]
       end
     end
-    setfield!(f, :matrix, matrix)
+    setfield!(a, :matrix, matrix)
   end
-  return f.matrix
+  return a.matrix
 end
 
 (h::FreeModuleHom)(a::FreeModElem) = image(h, a)
 
 @doc Markdown.doc"""
-    hom(F::FreeMod{T}, G, a::Union{Vector,MatElem{T}})
+    hom(F::FreeMod{T}, G, V::Union{Vector,MatElem{T}}) where {T}
 
-Return the morphism $F \to G$ where `F[i]` is mapped to `a[i]`.
+Create the homomorphism $F \to G$ defined by sending `F[i]` to `V[i]`.
```

Comment:
When `V` is a `MatElem{T}`, then what is `V[i]`? I am assuming it refers to either the i-th row or the i-th column, but which is it? Note that `V[i]` does not work in this case in general:
```
julia> A = fmpz_mat([ 1 2 ; 3 4 ])
[1   2]
[3   4]

julia> A[1]
ERROR: ArgumentError: linear indexing not supported for non-vector matrices
Stacktrace:
 [1] getindex(M::fmpz_mat, x::Int64)
   @ AbstractAlgebra.Generic ~/.julia/packages/AbstractAlgebra/yXuIT/src/generic/Matrix.jl:110
 [2] top-level scope
   @ REPL[47]:1

julia> A[1,:]
[1   2]

julia> A[:,1]
[1]
[3]
```